### PR TITLE
Extended syntax: Sharing

### PR DESCRIPTION
**Pull request by Anabra**
See #32

### CODE REVIEW

File: grin/test/AbstractInterpretation/ExtendedSyntax/SharingSpec.hs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            one <- pure (COne)
+            l0 <- store one
+            l1 <- store one
+            _1 <- update l0 one
+            _2 <- fun l0
+            v <- fetch l0
+            pure ()
+        |]
+    let result = calcSharedLocations code
+    let exptected = Set.fromList [0]
```
Comment (bjorn3) [inline grin/test/AbstractInterpretation/ExtendedSyntax/SharingSpec.hs]: *expected

File: grin/test/AbstractInterpretation/ExtendedSyntax/SharingSpec.hs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+calcSharingResult :: Exp -> SharingResult
+calcSharingResult prog
+  | (shProgram, shMapping) <- codeGen prog
+  , computer <- _airComp . evalAbstractProgram $ shProgram
+  , shResult <- toSharingResult shMapping computer
+  = shResult
+
+testProgram :: Exp
+-- The syntax conversion preserves the abstarct heap layout
```
Comment (bjorn3) [inline grin/test/AbstractInterpretation/ExtendedSyntax/SharingSpec.hs]: *abstract

Comment (luc-tielen) [reply grin/test/AbstractInterpretation/ExtendedSyntax/SharingSpec.hs]:
```suggestion
-- The syntax conversion preserves the abstract heap layout
```

File: grin/test/AbstractInterpretation/ExtendedSyntax/SharingSpec.hs (hunk 1 above)
Comment (Anabra) [reply grin/test/AbstractInterpretation/ExtendedSyntax/SharingSpec.hs]:
```suggestion
    let expected = Set.fromList [0]
```

**Pull request merged**